tunnels.go

```diff
@@ -0,0 +1,294 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/google/go-querystring/query"
+	"github.com/pkg/errors"
+)
+
+// Tunnel is the struct definition of a tunnel.
+type Tunnel struct {
+	ID          string             `json:"id,omitempty"`
+	Name        string             `json:"name,omitempty"`
+	Secret      string             `json:"tunnel_secret,omitempty"`
+	CreatedAt   *time.Time         `json:"created_at,omitempty"`
+	DeletedAt   *time.Time         `json:"deleted_at,omitempty"`
+	Connections []TunnelConnection `json:"connections,omitempty"`
+}
+
+// TunnelConnection represents the connections associated with a tunnel.
+type TunnelConnection struct {
+	ColoName           string `json:"colo_name"`
+	UUID               string `json:"uuid"`
+	IsPendingReconnect bool   `json:"is_pending_reconnect"`
+}
+
+// TunnelsDetailResponse is used for representing the API response payload for
+// multiple tunnels.
+type TunnelsDetailResponse struct {
+	Result []Tunnel `json:"result"`
+	Response
+}
+
+// TunnelDetailResponse is used for representing the API response payload for
+// a single tunnel.
+type TunnelDetailResponse struct {
+	Result Tunnel `json:"result"`
+	Response
+}
+
+type TunnelParams struct {
+	AccountID string
+	UUID      string
```

Comment:
It could be useful to say what this is

The same comment applies multiple times below (and it's always the same, the Tunnel ID)

Reply:
is it still UUID in the responses? I'm happy to swap to ID if that is what we are returning.